### PR TITLE
solving issues #21and #7

### LINKED ISSUE #7: user-service should not accept null user name
* In `users-service` `User.java` currently there is no validation for `name` field.
* Ther should be a Non-null validation check for this entity object.

### LINKED ISSUE #21: Implement DELETE house API
## Implement DELETE house API

* Users should be able to delete a house which was added.
* REST endpoint should be `/communitites/{communityId}/houses/{houseId}`
* HTTP method should be `DELETE`

### PR DESCRIPTION
**Pull request by Sountastic**

### CODE REVIEW

File: src/main/java/com/myhome/controllers/CommunityController.java [hunk 1]
```diff
@@ -20,160 +20,169 @@
...
+        var houseIds = communityService.addHousesToCommunity(communityId, communityHouses);
+        var response = new AddCommunityHouseResponse();
+        response.setHouses(houseIds);
+        return ResponseEntity.status(HttpStatus.CREATED).body(response);
+    }
+
+
+    @Operation(description = "Delete a house from the community given a community id and a house id")
+    @DeleteMapping(
+            path = "/communities/{communityId}/deleteHouse/{houseId}",
```
Comment (jmprathab) [inline src/main/java/com/myhome/controllers/CommunityController.java]: Following REST API best practices. Can the endpoint be `/communities/{communityId}/houses/{houseId}`?

**Review by jmprathab — CHANGES_REQUESTED** [review]
@Sountastic Please see my comment in `CommunityController.java`

As mentioned in README, we follow Square Java code style. If your time permits, please reformat your changes with code style from [https://github.com/square/java-code-styles](Square Java Styles)

File: src/main/java/com/myhome/controllers/CommunityController.java (hunk 1 above)
Comment (Sountastic) [reply src/main/java/com/myhome/controllers/CommunityController.java]: Changed ;)

**Comment by Sountastic** [comment]
@jmprathab by mistake i made a wrong commit -which deleted and paste a bunch of code in the community controller, I revert it and should be fine now

File: src/main/java/com/myhome/controllers/CommunityController.java [hunk 2]
```diff
@@ -206,4 +206,3 @@ public CommunityController(
     }
   }
 }
```
Comment (jmprathab) [inline src/main/java/com/myhome/controllers/CommunityController.java:238]: @Sountastic A corresponding `@DeleteMapping` has to be present in the controller. It seems to be missing.

**Review by jmprathab — CHANGES_REQUESTED** [review]
Please see my comments in `CommunityController.java`

File: src/main/java/com/myhome/controllers/CommunityController.java [hunk 3]
```diff
@@ -187,6 +187,21 @@ public CommunityController(
...
 
+  @Operation(description = "Deletion of house from the community given a community id and a house id")
+  @DeleteMapping(
+      path = "/communities/{communityId}/houses/{houseId}"
+  )
+  public ResponseEntity<Void> deleteCommunityHouse(
+      @PathVariable String communityId, @PathVariable String houseId
+  ) {
+    communityService.deleteHouseFromCommunityByHouseId(houseId);
+    communityService.deleteHouseFromCommunityByHouseId(houseId);
```
Comment (jmprathab) [inline src/main/java/com/myhome/controllers/CommunityController.java]: Looks like duplicated by accident?

**Comment by jmprathab** [comment]
@Sountastic Please resolve the conflict by pulling the latest changes. If changes are completed please resolve the comments.

**Review by Sountastic — COMMENTED** [review]
I pulled latest changes and there should be no conflicts,

**Review by jmprathab — APPROVED** [review]
Looks good.

**Pull request merged**